src/mem3/src/mem3_sync_event_listener.erl

```diff
@@ -301,30 +301,19 @@ get_state(Pid) ->
     Pid ! {get_state, Ref, self()},
     receive
         {Ref, State} -> State
-    after 10 ->
+    after 500 ->
         timeout
     end.
 
 
-wait_state_frequency(Pid, Val) ->
+wait_state(Pid, Field, Val) when is_pid(Pid), is_integer(Field) ->
+    
     WaitFun = fun() ->
         case get_state(Pid) of
-            timeout ->
-                wait;
-            #state{frequency = Val} ->
-                true
-        end
-    end,
-    test_util:wait(WaitFun).
-
-
-wait_state_delay(Pid, Val) ->
-    WaitFun = fun() ->
-        case get_state(Pid) of
-            timeout ->
-                wait;
-            #state{delay = Val} ->
-                true
+            #state{} = S when element(Field, S) == Val ->
```

Comment:
This is an elegant solution, and nearly as nice as the pattern match it replaces.